hunts/models.py

```diff
@@ -47,27 +48,72 @@ def get_num_unlocked(self):
     def get_num_metas_solved(self):
         return self.puzzles.filter(Q(status=Puzzle.SOLVED), Q(is_meta=True)).count()
 
-    def get_solves_per_hour(self):
-        current_time = timezone.now()
+    def get_num_metas_unsolved(self):
+        return self.puzzles.filter(~Q(status=Puzzle.SOLVED), Q(is_meta=True)).count()
+
+    # Gets the number of puzzles that are either solved or have a solved meta via DFS.
+    def get_progression(self):
+        solved_puzzles = self.puzzles.filter(status=Puzzle.SOLVED)
+        solved_puzzle_ids = {p.id for p in solved_puzzles}
+        puzzles_to_check = [p for p in solved_puzzles]
+
+        while len(puzzles_to_check) > 0:
+            next_puzzle = puzzles_to_check.pop()
+            if next_puzzle.is_meta:
+                for feeder in next_puzzle.feeders.all():
+                    if feeder.id not in solved_puzzle_ids:
+                        solved_puzzle_ids.add(feeder.id)
+                        puzzles_to_check.append(feeder)
+
+        return len(solved_puzzle_ids)
+
+    # Returns a list of solved meta names and solve times in [name, time] pairs.
+    # Solve times are given in dd hh:mm (am/pm) format (e.g., Fr 4:00 pm).
+    def get_meta_solve_list(self):
+        solved_metas = self.puzzles.filter(Q(status=Puzzle.SOLVED), Q(is_meta=True))
+        sorted_solved_metas = sorted(solved_metas, key=lambda x: x.solved_time())
```

Comment:
I think this could also make a bunch of db queries, probably fine for now but maybe add a TODO for it (I think we are going to want to prefetch before calling solved_time() on each meta since each call makes a new lookup)

Reply:
Turns out that I can use `.annotate(custom_condition_here)` in the Django ORM to replicate `solved_time()` without making too many DB queries from calling `solved_time()` over and over again! I've replaced every instance of calling the latter too many times with the former, which should hopefully amend any future problems.

Reply:
Nice, thank you!! .annotate is definitely how you are supposed to do this in Django, good find. I think this may be slightly different (looks like it looks at the max time of all guesses instead of all correct guesses) but that's not a big deal (though maybe we should add a comment or TODO, in case we ever go back to an answer queue world)

Reply:
Looks like there are `Case` and `When` functions that allow for conditionally choosing. I'll look into those to give us the functionality that we want, hopefully.